metagraph/src/annotation/taxonomy/taxonomic_db.cpp

```diff
@@ -0,0 +1,315 @@
+#include "taxonomic_db.hpp"
+
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <cmath>
+#include "common/serialization.hpp"
+#include "seq_io/sequence_io.hpp"
+#include "common/utils/string_utils.hpp"
+#include "common/unix_tools.hpp"
+
+#include "common/logger.hpp"
+
+
+namespace mtg {
+namespace annot {
+
+using mtg::common::logger;
+
+typedef TaxonomyDB::NormalizedTaxId NormalizedTaxId;
+
+void TaxonomyDB::dfs_statistics(const NormalizedTaxId &node, const ChildrenList &tree,
+                                   std::vector<NormalizedTaxId> &tree_linearization) {
+    node_to_linearization_idx[node] = tree_linearization.size();
+    tree_linearization.push_back(node);
+    uint64_t depth = 0;
+    for(const auto &child: tree[node]) {
+        dfs_statistics(child, tree, tree_linearization);
+        tree_linearization.push_back(node);
+        if (node_depth[child] > depth) {
+            depth = node_depth[child];
+        }
+    }
+    node_depth[node] = depth + 1;
```

Comment:
Is it possible to receive as input a tree where the distances leaf-root to be a constant value? In this case the current formula for finding node_depths can behave pretty weird

Reply:
I think for taxonomy trees they are not the same, are they?

Reply:
I think so, lets discuss at the meeting about the best approach here